File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.cpp
class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        int n=gain.size();
       int prev=0;
        int mx=0;
        for(auto &x:gain)
        {
            prev+=x;
            mx=max(prev,mx);
        }
        return mx;
    }
};

File: 1732-find-the-highest-altitude/1732-find-the-highest-altitude.js
/**
 * @param {number[]} gain
 * @return {number}
 */
var largestAltitude = function(gain) {
    //find the highest altitude
    let n=gain.length;
    let prev = 0;
    const res = [];
    res.push(0);
    for(let i=0;i<n;i++){
        prev = prev+gain[i];
        res.push(prev);
    }
    const ans = Math.max(...res);
    return ans;
};
-5,1,5,0,-7
-5,-4,1,1,-6